pennylane/templates/state_preparations/basis.py

```diff
@@ -103,8 +108,21 @@ def compute_decomposition(basis_state, wires):  # pylint: disable=arguments-diff
         [PauliX(wires=['a']),
         PauliX(wires=['b'])]
         """
-        op_list = []
-        for wire, state in zip(wires, basis_state):
-            if state == 1:
-                op_list.append(qml.PauliX(wire))
+        # if the state is not a JAX tracer or TensorFlow non-eager tensor, compute
+        # the decomposition using PauliX gates on the '1' wires
+        if not qml.math.is_abstract(basis_state):
```

Comment:
Why the decomposition depends on the type of the basis_state? I guess for tracers it is because you don't want to give parameters, but you we don't use this one for both cases?

Reply:
I'm reluctant to change existing behaviour if there's nothing wrong with it. And also because there will be unnecessary operators in cases where the basis state is close to being all zeros.

For tracers, since we don't know if the value of the basis state is 0 or 1, we need these extra operators just in case a 1 is passed in.